src/wallet/index.ts

```diff
@@ -281,6 +281,28 @@ export class Wallet {
 		}
 	}
 
+	/**
+	 * Stops the wallet. Use this method to prepare the wallet to be de
+	 * @returns {Promise<Result<string>>}
+	 */
+	public async stop(): Promise<Result<string>> {
+		try {
+			// if we are refreshing, we need to wait for it to finish
+			if (this.isRefreshing) {
+				await this.refreshWallet();
+			}
```

Comment:
Should this be a while loop instead waiting for `isRefreshing` to be false? Or use `await this.refreshWallet({ force: true });`, otherwise this reads like it just adds another refresh to the queue?

Reply:
There is no queue. If you call refreshWallet twice, second one will just wait for the first one to finish and will not run refresh logic again

Reply:
Ok I wasn't sure how to read it